src/postselection.jl

```diff
@@ -141,98 +127,116 @@ function initByMsmt_2D(data,Anum,numSegments,numMeas_A,indMeas,selectMeas,select
 
   ind0=indMeas[1,:]
   ind1=indMeas[2,:]
+  nbins = 500
+
+  if ~isa(Anum, Array)
+    Anum = [Anum]
+  end
+  if ~isa(threshold, Array)
+    threshold = repmat([threshold],length(Anum), 1)
+  end
 
-  data_A = data[Anum]
-  num_data_meas = length(data)-1
+  data_A = zeros(length(data[1]), length(Anum))
+  for (k, Aind) in enumerate(Anum)
+    data_A[:, k] = real(data[Aind])
+  end
+  num_data_meas = length(data)-length(Anum)
   data_D = Dict()
   datamat = Dict()
   var_con = Dict()
   data_con = Dict()
   ind=1
   for kk = 1:num_data_meas
-    if ind == Anum
+    while ind in Anum
```

Comment:
what is the point of this loop? Either there is a mistake here, or you simply want
```
ind = Anum[end] + 1
```